Sources/SwiftGRPCNIO/ClientCalls/ClientStreamingClientCall.swift

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import SwiftProtobuf
+import NIO
+
+/// A client-streaming gRPC call.
+///
+/// Messages should be sent via the `send` method; an `.end` message should be sent
+/// to indicate the final message has been sent.
+///
+/// The following futures are available to the caller:
+/// - `initialMetadata`: the initial metadata returned from the server,
+/// - `response`: the response from the call,
+/// - `status`: the status of the gRPC call,
+/// - `trailingMetadata`: any metadata returned from the server alongside the `status`.
+public class ClientStreamingClientCall<RequestMessage: Message, ResponseMessage: Message>: BaseClientCall<RequestMessage, ResponseMessage>, StreamingRequestClientCall, UnaryResponseClientCall {
+  public var response: EventLoopFuture<ResponseMessage> {
+    // It's okay to force unwrap because we know the handler is holding the response promise.
```

Comment:
The `clientChannelHandler` holds a strong reference to the `responseObserver` which in turn holds the promise so we shouldn't hold a strong reference to the promise here to avoid a retain cycle.

We could hold an `unowned` reference to the future since we know that the handler, observer and promise will live for at least as long as the `*Call`. What do you think?

Reply:
Sorry, but how is this a retain cycle? The promise only holds the response message, which shouldn't hold anything else. I'm probably missing something.

Reply:
Agh, it's not. I'm not sure what I was thinking before!

Reply:
I guess we could get rid of the `unowned` then?